### PR TITLE
Remove subsystem_labels option from DynamicsBackend

### LINKED ISSUE #235: Transpile problem using DynamicsBackend code
### Informations

- **Qiskit Dynamics version**: 0.4.1
- **Python version**: 3.9.15
- **Operating system**: Windows

### What is the current behavior?
Sorry to my English. Please understand this post.
When I use DynamicsBackend class at fakeprovider, there is variable 'subsystem_list'. Qiskit Document explanation said that 'subsystem_list' means the qubit which i wanted to use. 

There are problems. When I use pulse class, If I write 1 or another number except 0 at subsystem_list, there is error.
Also, If I write a list like [0,1], and If I use pulse class at two qubit, there is error.
I described all code below.

### Steps to reproduce the problem
I will write my code.

First problem:
from qiskit.providers.fake_provider import FakeKolkata
from qiskit_dynamics.backend import DynamicsBackend
backend= DynamicsBackend.from_backend(FakeKolkata(), subsystem_list=[1])
X_cal=QuantumCircuit(1)
X_cal.x(0)
X_cal.measure_all()
X_cal.calibrations
with pulse.build(backend, name='X') as x_q0:
    pulse.play(Drag(duration=160, beta=0.7171287840008055, amp=0.16, sigma=40), pulse.DriveChannel(0))
X_cal.add_calibration('x',[0], x_q0)
circ_transpile = transpile(X_cal, backend)

and there is error 
QiskitError: "Cannot unroll the circuit to the given basis, ['measure']. Instruction measure not found in equivalence library and no rule found to expand."

But if i write at subsystem_list = [0], there is not problem.

Second problem:
from qiskit.providers.fake_provider import FakeKolkata
from qiskit_dynamics.backend import DynamicsBackend
backend= DynamicsBackend.from_backend(FakeKolkata(), subsystem_list=[0,1])
circ=QuantumCircuit(2)
circ.h(0)
circ.x(0)
circ.z(1)
circ.cx(0,1)
circ.y(0)
circ.y(1)
circ.measure_all()
with pulse.build(backend, name='X') as x_q0:
    pulse.play(Drag(duration=160, beta=-1.5, amp=0.15575757575757576, sigma=40), pulse.DriveChannel(0))
circ.add_calibration('x',[0], x_q0)
with pulse.build(backend, name='Y') as y_q0:
    pulse.shift_phase(np.pi,pulse.DriveChannel(0))
    pulse.play(Drag(duration=160, beta=-1.5, amp=0.15575757575757576, sigma=40), pulse.DriveChannel(0))
circ.add_calibration('y',[0], y_q0)
with pulse.build(backend, name='Y') as y_q1:
    pulse.shift_phase(np.pi,pulse.DriveChannel(1))
    pulse.play(Drag(duration=160, beta=-1.5, amp=0.15575757575757576, sigma=40), pulse.DriveChannel(1))
circ.add_calibration('y',[1], y_q1)
circ_transpile = transpile(circ, backend)

and there is problem
TranspilerError: "Unable to translate the operations in the circuit: ['cx', 'z', 'h', 'measure', 'barrier'] to the backend's (or manually specified) target basis: ['barrier', 'measure', 'snapshot']. This likely means the target basis is not universal or there are additional equivalence rules needed in the EquivalenceLibrary being used. For more details on this error see: https://qiskit.org/documentation/stubs/qiskit.transpiler.passes.BasisTranslator.html#translation_errors"

### What is the expected behavior?

First case is just drive qubit 1 and yield the result. 
Second case also just drive qubit to use our calibrated X gate and Y gate and other qiskit own calibrated gate.

### Suggested solutions
Add all our calibration gate to basis gate.

### PR DESCRIPTION
**Pull request by DanPuzzuoli**
### Summary

Closes #235

### Details and comments (out of date, see next comment)

This is currently a draft to prototype a potential solution to #235, in which the `subsystem_labels` option is removed, and all qubits removed via the `subsystem_list` argument of `from_backend` are actually "kept" as trivial 1-dimensional systems.

So far:
- I've commented out all lines/blocks using `subsystem_labels`.
- Where necessary, new code has been added, e.g. construction of `subsystem_dims` in `from_backend` has been changed to include all pre-existing subsystems, but setting the dimension of the "removed" subsystems to `1` (instead of not including them at all).

Initial testing in a jupyter notebook seems to indicate that this behaves as expected:
- I can do `from_backend` with a non-trivial `subsystem_list`, and the backend is built without issue.
- I haven't yet done any verifications of correctness, but it seems like we can run sims and compute measurement results as well.

To do:
- Add number of qubits to the backend wherever it makes sense.
- Do some initial verification that the calculations are correct.
- The `subsystem_labels` option was used to raise errors of non-existent subsystems were measured. As this no longer exists, it would be nice to still warn users if their schedule measures a trivial subsystem. 
    - This could just be done by checking the dimension of the measured subsystem.
- Pre-existing tests need to be modified.

Question:
- This will remove the `subsystem_labels` option. I'm not sure how this could be deprecated, because it will literally serve no purpose and have no effect anymore.

### CODE REVIEW

**Comment by DanPuzzuoli** [comment]
I've now implemented all of the above changes. I'm going to now read through the existing tests to make sure it's covering the latest changes are adequately tested.

File: qiskit_dynamics/backend/dynamics_backend.py [hunk 1]
```diff
@@ -109,9 +110,7 @@ class DynamicsBackend(BackendV2):
       indicating solver methods and options. Defaults to the empty dictionary ``{}``.
     * ``subsystem_dims``: Dimensions of subsystems making up the system in ``solver``. Defaults to
       ``[solver.model.dim]``.
-    * ``subsystem_labels``: Integer labels for subsystems. Defaults to ``[0, ...,
-      len(subsystem_dims) - 1]``.
-    * ``meas_map``: Measurement map. Defaults to ``[[idx] for idx in subsystem_labels]``.
+    * ``meas_map``: Measurement map. Defaults to ``[[idx] for idx in range(len(subsystem_dims))]``.
```
Comment (to24toro) [inline qiskit_dynamics/backend/dynamics_backend.py:113]: [Q] What is `meas_map` in dynamicsbackend used for ?

File: test/dynamics/backend/test_dynamics_backend.py [hunk 2]
```diff
@@ -127,7 +127,18 @@ def test_measure_nonexistant_subsystem(self):
...
+            self.simple_backend.run(schedule)
+
+    def test_measure_trivial_subsystem(self):
+        """Attempt to measure subsystem with dimension 1."""
+
+        with pulse.build() as schedule:
+            pulse.play(pulse.Waveform([0.5, 0.5, 0.5]), pulse.DriveChannel(0))
+            pulse.acquire(duration=1, qubit_or_channel=1, register=pulse.MemorySlot(0))
+
+        self.simple_backend.set_options(subsystem_dims=[2, 1])
```
Comment (to24toro) [inline test/dynamics/backend/test_dynamics_backend.py:140]: [Q] What situation is the case that specifying explicitly one of subsystem_dims is 1?

File: qiskit_dynamics/backend/dynamics_backend.py (hunk 1 above)
Comment (DanPuzzuoli) [reply qiskit_dynamics/backend/dynamics_backend.py:113]: I always forget exactly how `meas_map` works. `DynamicsBackend` itself doesn't use `meas_map`, but I think it's required by transpilation and/or pulse schedule validation. I believe the default `meas_map` described here allows measurements of any subset of qubits to pass the validation checks.

File: test/dynamics/backend/test_dynamics_backend.py (hunk 2 above)
Comment (DanPuzzuoli) [reply test/dynamics/backend/test_dynamics_backend.py:140]: I wouldn't expect a user to set this themselves, but I wanted to have an explicit test case for this.

File: qiskit_dynamics/backend/dynamics_backend.py (hunk 1 above)
Comment (to24toro) [reply qiskit_dynamics/backend/dynamics_backend.py:113]: The `meas_map` is used for mapping qubits which have to be measured concurrently. The configuration of `meas_map` used here indicates each qubit can be measured independently.

Just asking, I just understood the need for transpile and pulse validation. Thanks.

File: test/dynamics/backend/test_dynamics_backend.py (hunk 2 above)
Comment (to24toro) [reply test/dynamics/backend/test_dynamics_backend.py:140]: If users use like 
```python
backend= DynamicsBackend.from_backend(FakeKolkata(), subsystem_list=[1])
```
do they observe the warning?
If so, I'm not sure why this is necessary.

Comment (DanPuzzuoli) [reply test/dynamics/backend/test_dynamics_backend.py:140]: The warning will be raised if they do
```
backend= DynamicsBackend.from_backend(FakeKolkata(), subsystem_list=[1])
```
then submit a schedule or circuit that measures qubit 0. I was imagining in this case that we should let users know that they're measuring a system that has no evolution (which I thought could confuse users if no warning were present).

**Pull request merged**